File: abc/abc325/abc325_b/abc325_b.cs
class A
{
  static int Max(int a, int b)
  {
    return (a > b ? a : b);
  }

  static void Main()
  {
    int n = int.Parse(Console.ReadLine());
    
    int[] cnt = new int[24];
    for(int i = 0; i < 24; i++)
    {
      cnt[i] = 0;
    }

    for(int i = 0; i < n; i++)
    {
      string[] t = Console.ReadLine().Split(" ");
      int w = int.Parse(t[0]);
      int x = int.Parse(t[1]);
      cnt[x] += w;
    }

    int ans = 0;
    for(int i = 0; i < 24; i++)
    {
      int sum = 0;

      for(int j = 0; j < 9; j++)
      {
        int k = (i + j) % 24;
        sum += cnt[k];
      }

      ans = Max(ans, sum);
    }

    Console.Write(ans);
  }
}

File: abc/abc325/abc325_b/abc325_b.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
  int n;
  cin >> n;

  vector<int> cnt(24, 0);
  for(int i = 0; i < n; i++) {
    int w, x;
    cin >> w >> x;
    cnt[x] += w;
  }

  int ans = 0;
  for(int i = 0; i < 24; i++) {
    int c = 0;
    for(int j = 0; j < 9; j++) {
      int t = (i + j) % 24;
      c += cnt[t];
    }

    ans = max(ans, c);
  }


  cout << ans;
}